frontend/dockerfile/instructions/commands.go

```diff
@@ -254,6 +253,12 @@ func (c *AddCommand) Expand(expander SingleWordExpander) error {
 	}
 	c.Chown = expandedChown
 
+	expandedChecksum, err := expander(c.Checksum)
```

Comment:
I think it is better if you keep the `Checksum` as `digest.Digest`. In here after expansion, you should call `digest.Parse()` to convert the string to `Digest` and error if the value is invalid.

Reply:
Unfortunately it is not possible to store `AddCommand`'s `Checksum` as `digest.Digest` (I can only try to do validation without type cast inside Expand, but it looks quite misplaced) as we set it before expansion. Sorry if I misunderstood your suggestion.
I modified the code and now convert (and check) during filling of `copyConfig`.